Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -0,0 +1,767 @@
+      refine ⟨Ioo (max a (t₁ - ε)) (min b (t₁ + ε)),
+        subset_inter
+          (fun t ht => @heqon t <| mem_of_mem_of_subset ht <| Ioo_subset_Ioo (by simp) (by simp))
+          (Ioo_subset_Ioo (by simp) (by simp)),
+        isOpen_Ioo, ?_⟩
+      rw [mem_Ioo]
+      exact ⟨max_lt ht₁.2.1 (by simp [hε]), lt_min ht₁.2.2 (by simp [hε])⟩
+  intros t ht
+  exact mem_setOf.mp ((subset_def ▸ hsub) t ht).1
+
+theorem isIntegralCurveOn_Ioo_eqOn_of_contMDiff_boundaryless [BoundarylessManifold I M]
+    (ht₀ : t₀ ∈ Ioo a b)
+    (hv : ContMDiff I I.tangent 1 (fun x => (⟨x, v x⟩ : TangentBundle I M)))
+    (hγ : IsIntegralCurveOn γ v (Ioo a b)) (hγ' : IsIntegralCurveOn γ' v (Ioo a b))
+    (h : γ t₀ = γ' t₀) : EqOn γ γ' (Ioo a b) :=
+  isIntegralCurveOn_Ioo_eqOn_of_contMDiff
+    ht₀ (fun _ _ => BoundarylessManifold.isInteriorPoint I) hv hγ hγ' h
+
+/-- Global integral curves are unique.
+
+  If a continuously differentiable vector field `v` admits two global integral curves
+  `γ γ' : ℝ → M`, and `γ t₀ = γ' t₀` for some `t₀`, then `γ` and `γ'` are equal. -/
+theorem isIntegralCurve_eq_of_contMDiff (hγt : ∀ t, I.IsInteriorPoint (γ t))
+    (hv : ContMDiff I I.tangent 1 (fun x => (⟨x, v x⟩ : TangentBundle I M)))
+    (hγ : IsIntegralCurve γ v) (hγ' : IsIntegralCurve γ' v) (h : γ t₀ = γ' t₀) : γ = γ' := by
+  ext t
+  obtain ⟨T, hT, ht⟩ : ∃ T > 0, t ∈ Ioo (t₀ - T) (t₀ + T) := by
+    refine ⟨2 * |t - t₀| + 1, add_pos_of_nonneg_of_pos (by simp) zero_lt_one, ?_⟩
+    rw [mem_Ioo]
+    by_cases ht : t - t₀ < 0
+    · rw [abs_of_neg ht]
+      constructor <;> linarith
+    · rw [abs_of_nonneg (not_lt.mp ht)]
+      constructor <;> linarith
+  exact isIntegralCurveOn_Ioo_eqOn_of_contMDiff
+    (Real.ball_eq_Ioo t₀ T ▸ Metric.mem_ball_self hT) (fun t _ => hγt t) hv
+    (IsIntegralCurveOn.mono (hγ.isIntegralCurveOn _) (subset_univ _))
+    (IsIntegralCurveOn.mono (hγ'.isIntegralCurveOn _) (subset_univ _)) h ht
+
+theorem isIntegralCurve_Ioo_eq_of_contMDiff_boundaryless [BoundarylessManifold I M]
+    (hv : ContMDiff I I.tangent 1 (fun x => (⟨x, v x⟩ : TangentBundle I M)))
+    (hγ : IsIntegralCurve γ v) (hγ' : IsIntegralCurve γ' v) (h : γ t₀ = γ' t₀) : γ = γ' :=
+  isIntegralCurve_eq_of_contMDiff
+    (fun _ => BoundarylessManifold.isInteriorPoint I) hv hγ hγ' h
+
+lemma exists_isIntegralCurveOn_Ioo_eqOn [BoundarylessManifold I M]
+    (hv : ContMDiff I I.tangent 1 (fun x => (⟨x, v x⟩ : TangentBundle I M))) {x : M}
+    (h : ∀ a, ∃ γ, γ 0 = x ∧ IsIntegralCurveOn γ v (Ioo (-a) a)) {a a' : ℝ} (hpos : 0 < a')
+    (hle : a' ≤ a) : EqOn (choose (h a')) (choose (h a)) (Ioo (-a') a') := by
+  have ⟨h0', hγ'⟩ := choose_spec (h a')
+  have ⟨h0, hγ⟩ := choose_spec (h a)
+  apply isIntegralCurveOn_Ioo_eqOn_of_contMDiff_boundaryless _ hv hγ' (hγ.mono _) (by rw [h0', h0])
+  · rw [mem_Ioo]
+    exact ⟨neg_lt_zero.mpr hpos, by positivity⟩
+  · apply Ioo_subset_Ioo <;> linarith
+
+/-- Auxiliary lemma. Suppose for every `a : ℝ`, there exists an integral curve defined on
```

Comment:
How about this? (You can add "aux" to the lemma name if you want.)

/-- Suppose for every `a : ℝ`, there exists an integral curve `γ a` on `Ioo (-a) a`.
Then, the global curve `γ_ext := fun t ↦ γ (|t| + 1) t` agrees with each `γ a` on `Ioo (-a) a`.
This will help us show that `γ` is a global integral curve. -/